qiskit/primitives/statevector_sampler.py

```diff
@@ -0,0 +1,215 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023, 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Statevector Sampler class
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Iterable
+
+import numpy as np
+from numpy.typing import NDArray
+
+from qiskit import ClassicalRegister, QiskitError, QuantumCircuit
+from qiskit.circuit import ControlFlowOp
+from qiskit.quantum_info import Statevector
+
+from .base import BaseSamplerV2
+from .base.validation import _has_measure
+from .containers import (
+    BitArray,
+    PrimitiveResult,
+    PubResult,
+    SamplerPub,
+    SamplerPubLike,
+    make_data_bin,
+)
+from .containers.bit_array import _min_num_bytes
+from .primitive_job import PrimitiveJob
+from .utils import bound_circuit_to_instruction
+
+
+@dataclass
+class _MeasureInfo:
+    creg_name: str
+    num_bits: int
+    num_bytes: int
+    qreg_indices: list[int]
+
+
+class Sampler(BaseSamplerV2):
+    """
+    Simple implementation of :class:`BaseSamplerV2` with Statevector.
+    """
+
+    _DEFAULT_SHOTS: int = 512
+
+    def __init__(self, *, seed: np.random.Generator | int | None = None):
+        """
+        Args:
+            seed: The seed for random number generator.
+        """
```

Comment:
I would prefer a `shots` or `default_shots` to be an init kwarg so a user has a non-internal way of changing it.
```suggestion
    def __init__(self, *, shots: int = 1024, seed: np.random.Generator | int | None = None):
        """
        Args:
            shots: The default shots for the sampler if not specified during run.
            seed: The seed or Generator object for random number generation.
                If None a random seeded default RNG will be used.
        """
```
Then you can also add a `shots` or `default_shots` property like you have for seed.